plugins/openvr/src/ViveControllerManager.cpp

```diff
@@ -262,6 +428,311 @@ bool ViveControllerManager::isHeadControllerMounted() const {
+
+    // 1st
+    glm::vec3 palmFacing = glm::normalize(glm::cross(pinkie1 - wristPos, thumb2 - wristPos));
+    glm::vec3 handForward = glm::normalize(point1 - wristPos);
+    glm::vec3 x = glm::normalize(glm::cross(palmFacing, handForward));
+    glm::vec3 y = glm::normalize(point2 - point1);
+    glm::vec3 z = (hand == 0) ? glm::cross(y, x) : glm::cross(x, y);
+    glm::mat4 rotMat1 = glm::mat4(glm::vec4(x, 0.0f),
+                                  glm::vec4(y, 0.0f),
+                                  glm::vec4(z, 0.0f),
+                                  glm::vec4(0.0f, 0.0f, 0.0f, 1.0f));
+    glm::quat rot1 = glm::normalize(glmExtractRotation(rotMat1));
+    if (!isNaN(rot1)) {
+        _inputDevice->_poseStateMap[joint1] = controller::Pose(point1, rot1);
+    }
+
+
+    // 2nd
+    glm::vec3 x2 = x; // glm::normalize(glm::cross(point3 - point2, point2 - point1));
+    glm::vec3 y2 = glm::normalize(point3 - point2);
+    glm::vec3 z2 = (hand == 0) ? glm::cross(y2, x2) : glm::cross(x2, y2);
+
+    glm::mat4 rotMat2 = glm::mat4(glm::vec4(x2, 0.0f),
+                                  glm::vec4(y2, 0.0f),
+                                  glm::vec4(z2, 0.0f),
+                                  glm::vec4(0.0f, 0.0f, 0.0f, 1.0f));
+    glm::quat rot2 = glm::normalize(glmExtractRotation(rotMat2));
+    if (!isNaN(rot2)) {
+        _inputDevice->_poseStateMap[joint2] = controller::Pose(point2, rot2);
+    }
+
+
+    // 3rd
+    glm::vec3 x3 = x; // glm::normalize(glm::cross(point4 - point3, point3 - point1));
+    glm::vec3 y3 = glm::normalize(point4 - point3);
+    glm::vec3 z3 = (hand == 0) ? glm::cross(y3, x3) : glm::cross(x3, y3);
+
+    glm::mat4 rotMat3 = glm::mat4(glm::vec4(x3, 0.0f),
+                                  glm::vec4(y3, 0.0f),
+                                  glm::vec4(z3, 0.0f),
+                                  glm::vec4(0.0f, 0.0f, 0.0f, 1.0f));
+    glm::quat rot3 = glm::normalize(glmExtractRotation(rotMat3));
+    if (!isNaN(rot3)) {
+        _inputDevice->_poseStateMap[joint3] = controller::Pose(point3, rot3);
+    }
+
+
+    // 4th
+    glm::quat rot4 = rot3;
+    if (!isNaN(rot4)) {
+        _inputDevice->_poseStateMap[joint4] = controller::Pose(point4, rot4);
+    }
+}
+
+
+void ViveControllerManager::updateCameraHandTracker(float deltaTime,
+                                                    const controller::InputCalibrationData& inputCalibrationData) {
+
+    if (areBothHandControllersActive(_system)) {
+        // if both hand-controllers are in use, don't do camera hand tracking
+        disableGestureDetection();
+    } else {
+        enableGestureDetection();
+    }
+
+    if (!_viveCameraHandTracker) {
+        return;
+    }
+
+    const GestureResult* results = NULL;
+    int handTrackerFrameIndex { -1 };
+    int resultsHandCount = GetGestureResult(&results, &handTrackerFrameIndex);
+
+    if (handTrackerFrameIndex >= 0 /* && handTrackerFrameIndex != _lastHandTrackerFrameIndex */) {
```

Comment:
why is this check commented out?

Reply:
Will add a `// FIXME` comment to address later.